### PR TITLE
Bump version and docs for 2.1.X

### PR DESCRIPTION
**Pull request by wmaxey**

### CODE REVIEW

File: docs/releases/changelog.md [hunk 1]
```diff
@@ -10,6 +10,64 @@ It pulls in the latest version of upstream libc++ and marks the beginning of
...
+
+We are excited to announce the release of libcudacxx 2.1. While there are no breaking changes in this release, we are increasing the semantic major version to better synchronize our release versions with [Thrust](https://github.com/NVIDIA/thrust) and [CUB](https://github.com/NVIDIA/cub) repositories. This effort aims to better unify our libraries, and ensure that users can more easily determine which versions of each library are compatible with one another. We are making this change because we want to start thinking about these three projects not as independent libraries, but as three parts of a single, [Core CUDA C++ Library (CCCL)](https://github.com/NVIDIA/cccl). In the near future, before the next release, we plan to merge the three libraries into a single monorepository at [nvidia/cccl](https://github.com/NVIDIA/cccl). We believe that this consolidation will make it even easier for users to take advantage of the powerful features and optimizations available in our CUDA libraries.
+
+Supported ABI Versions: 4 (default), 3, and 2.
+
+### New Features
+
+- #313: Add `cuda/std/span`.
+- #299: Add `mdspan` reference implementation.
```
Comment (miscco) [inline docs/releases/changelog.md:24]:
```suggestion
- #299: Add `cuda/std/mdspan` reference implementation.
```

File: docs/releases/changelog.md [hunk 2]
```diff
@@ -10,6 +10,64 @@ It pulls in the latest version of upstream libc++ and marks the beginning of
...
+  - Thanks Yu You for this contribution.
+- #349: Adds concepts and several backports of ranges' CPOs to C++14.
```
Comment (miscco) [inline docs/releases/changelog.md:26]:
```suggestion
- #349: Add `cuda/std/concepts` and backports them to C++14. You will be able to utilize C++20 concepts already in C++14/17 through existing SFINAE techniques.
```

File: docs/releases/changelog.md [hunk 3]
```diff
@@ -10,6 +10,64 @@ It pulls in the latest version of upstream libc++ and marks the beginning of
...
+
```
Comment (miscco) [inline docs/releases/changelog.md:27]:
```suggestion
 - #333: Add support for structured bindings for `cuda::std::tuple`, `cuda::std::pair` and `cuda::std::array`
 
```

File: docs/releases/changelog.md [hunk 4]
```diff
@@ -10,6 +10,64 @@ It pulls in the latest version of upstream libc++ and marks the beginning of
...
+- #369: Fix cuda::atomic_thread_fence example.
+- #110: Fixed rename errors.
+- #353: Update thread scope for synchronization domains.
+- #367: Fix warnings in several headers.
+- #379: Update atomics backend to be compatible with `atomic_ref`.
+- #378: Prevent conflict with MSVC macros in 14.35.
+- #355: Modularize `<functional>`.
+- #309: Implement `{async_}resource_ref`.
+- #372: Improve handling of internal headers and modularization.
+- #381: Drop libcxx testing from DVS.
```
Comment (miscco) [inline docs/releases/changelog.md:67]: This is a purely internal feature, so we should drop it
```suggestion
```

**Review by miscco — CHANGES_REQUESTED** [review]

File: docs/releases/changelog.md [hunk 5]
```diff
@@ -10,6 +10,64 @@ It pulls in the latest version of upstream libc++ and marks the beginning of
...
+
+Adds `<cuda/std/span>`, `<cuda/std/mdspan>`, and `<cuda/std/concepts>` to libcu++.
+
+We are excited to announce the release of libcudacxx 2.1. While there are no breaking changes in this release, we are increasing the semantic major version to better synchronize our release versions with [Thrust](https://github.com/NVIDIA/thrust) and [CUB](https://github.com/NVIDIA/cub) repositories. This effort aims to better unify our libraries, and ensure that users can more easily determine which versions of each library are compatible with one another. We are making this change because we want to start thinking about these three projects not as independent libraries, but as three parts of a single, [Core CUDA C++ Library (CCCL)](https://github.com/NVIDIA/cccl). In the near future, before the next release, we plan to merge the three libraries into a single monorepository at [nvidia/cccl](https://github.com/NVIDIA/cccl). We believe that this consolidation will make it even easier for users to take advantage of the powerful features and optimizations available in our CUDA libraries.
+
+Supported ABI Versions: 4 (default), 3, and 2.
+
+### New Features
+
+- #313: Add `cuda/std/span`.
```
Comment (miscco) [inline docs/releases/changelog.md:23]:
```suggestion
- #313: Add `cuda/std/span` and backport it to C++14
```

File: docs/releases/changelog.md [hunk 6]
```diff
@@ -10,6 +10,71 @@ It pulls in the latest version of upstream libc++ and marks the beginning of
...
+- #349: Add `cuda/std/concepts` and backport them to C++14. You will be able to utilize C++20 concepts already in C++14/17 through existing SFINAE techniques.
+
+### Issues Fixed
+
+- #328: Docs: Fix broken links.
+  - Thanks chaosink for this contribution.
+- #329: Block local memory tests on pascal.
+- #332: Add support for clang-15.
+- #317: Fix structured binding support.
+- #333: Fix tuple structured bindings.
```
Comment (wmaxey) [inline docs/releases/changelog.md:37]:
```suggestion
```

File: docs/releases/changelog.md [hunk 7]
```diff
@@ -10,6 +10,64 @@ It pulls in the latest version of upstream libc++ and marks the beginning of
...
+- #107: Description of policy for backporting Standard C++ features.
+- #128: Fix a typo in the `cuda::` heterogeneous policy.
+- #369: Fix cuda::atomic_thread_fence example.
+- #110: Fixed rename errors.
+- #353: Update thread scope for synchronization domains.
+- #367: Fix warnings in several headers.
+- #379: Update atomics backend to be compatible with `atomic_ref`.
+- #378: Prevent conflict with MSVC macros in 14.35.
+- #355: Modularize `<functional>`.
+- #309: Implement `{async_}resource_ref`.
```
[the remark this answers is not in the document]
Comment (wmaxey) [reply docs/releases/changelog.md:67]: We need to have an entry for the PR unless it is a docs change. Mentioning it as internal/experimental is fine.

```suggestion
- #309: Add an experimental `{async_}resource_ref`.
```

[the remark this answers is not in the document]
Comment (miscco) [reply docs/releases/changelog.md:67]: 👍

**Review by miscco — APPROVED** [review]

**Review by jrhemstad — APPROVED** [review]

**Pull request merged**